packages/cli/src/UserManagement/PermissionChecker.ts

```diff
@@ -101,15 +99,40 @@ export class PermissionChecker {
 			policy = 'workflowsFromSameOwner';
 		}
 
+		const parentWorkflowOwner =
+			await Container.get(OwnershipService).getWorkflowOwnerCached(parentWorkflowId);
+
 		const subworkflowOwner = await Container.get(OwnershipService).getWorkflowOwnerCached(
 			subworkflow.id,
 		);
 
-		const errorToThrow = new SubworkflowOperationError(
-			`Target workflow ID ${subworkflow.id ?? ''} may not be called`,
-			subworkflowOwner.id === userId
+		let parentWorkflow: WorkflowEntity;
+
+		try {
+			parentWorkflow = await Container.get(WorkflowRepository).findOneOrFail({
+				where: { id: parentWorkflowId },
+			});
+		} catch (error) {
+			throw new ApplicationError('Failed to find parent workflow', { extra: { parentWorkflowId } });
+		}
+
+		const executeWorkflowNode = Object.values(parentWorkflow.nodes).find(
```

Comment:
The problem I see here is that if a workflow contains more than 1 execute workflow node, we'll get the first one we find.

Instead, what we could do is pass the `node` from https://github.com/n8n-io/n8n/blob/f5502cc628f6b348f7fe3325b96ec9dc3360beaf/packages/core/src/NodeExecuteFunctions.ts#L3155 to this function call. Then we'd have the exact node that was trying to execute a child workflow and display the error in the proper place.